js/analyzer.js

```diff
@@ -1,65 +1,232 @@
 /**
- * Created by danny on 3/19/15.
+ * Text Analyzer, accepts args for config and
+ * @param args
+ * @constructor
  */
+Analyzer = function (args){
+    this.config = args;
+    this.init();
+};
 
-/*text preprocessor*/
+/**
+ * Analyzer initialisation. Loads defaults and overloads custom settings.
+ */
+Analyzer.prototype.init = function() {
 
+    //init preprocessor if not exists
+    if (typeof yst_pp != 'object' || yst_pp.inputText != this.config.textString) {
+        yst_pp = new PreProcessor(this.config.textString);
+    }
+    //if no available function queues, make new queue
+    if (typeof this.queue != 'Array') {
```

Comment:
I would make a function of line 20-29 to set the queue, because it's a single responsibility